libs/filter_lib/src/graph_utils.rs

```diff
@@ -57,61 +57,56 @@ pub fn generate_target_graph(vertices: Vec<String>,
+    for node_str in nodes_iterator {
+        // 1. Is this node already in the graph?  If so, ignore it and move "up" the tree
+        if !node_str_to_node_handle.contains_key(node_str) {
+            let new_node_handle = graph.add_node(node_str.to_string());
+            node_str_to_node_handle.insert(node_str.to_string(), new_node_handle);
+            graph.add_edge(prev_node_handle, new_node_handle, "".to_string());
+            prev_node_handle = new_node_handle;
+        }
+        else {
+            prev_node_handle = node_str_to_node_handle[node_str];
+        }
     }
     graph
 }
 
-
-
-/* Note:  the more efficient algorithm to do this (that is also used by the boost library) is here:
- * https://citeseerx.ist.psu.edu/viewdoc/download;jsessionid=E6BEA4B7B3694938A0BBEBB3604F14C7?doi=10.1.1.101.5342&rep=rep1&type=pdf
- * But that's more complicated than we need right now for a prototype, so the below algorithm 
- * does subgraph isomorphism only on non-branching trees.  So... they're always subgraph isomorphic, unless target is bigger than trace.
- * 
- * All this algorithm does is check the length, and if so, create a mapping between the trace and target graphs
- * Arguments:
- * @trace_graph: the graph of the trace observed
- * @target_graph: the graph of the target pattern we want to match to
- *
- * Return value:
- * @mapping: a hashmap mapping vertices in target_graph to those in trace_graph if the graphs are subgraph isomophic, and
- *           an empty hashmap otherwise
- */
-pub fn get_sub_graph_mapping(trace_graph:  Graph<String, String>, target_graph: Graph<String, String>) -> HashMap<NodeIndex, NodeIndex> {
-    // Right now, simply having more nodes than the target will be sufficient to say yes because
-    // I haven't implemented branching.  So that's what we're going to do, and we'll make this more general later
-    let mut mapping = HashMap::new();
-    if trace_graph.node_count() >= target_graph.node_count() {
-        let trace_graph_order = toposort(&trace_graph, None).unwrap();
-        let target_graph_order = toposort(&target_graph, None).unwrap();
-        let trace_root = trace_graph_order[0];
-        let target_root = target_graph_order[0];
-        mapping.insert(trace_root, target_root);
-        let mut trace_children: Vec<NodeIndex> = trace_graph.neighbors(trace_root).collect();
-        let mut target_children: Vec<NodeIndex> = trace_graph.neighbors(target_root).collect();
-        while trace_children.len() != 0 && target_children.len() != 0 {
-            let trace_child = trace_children[0];
-            let target_child = target_children[0];
-            mapping.insert(target_child, trace_child);
-            trace_children = trace_graph.neighbors(trace_child).collect();
-            target_children = trace_graph.neighbors(target_child).collect();
+pub fn get_node_with_id(graph: &Graph<String, String>, node_name: String) -> Option<NodeIndex> {
```

Comment:
Is this function used? Who uses it?

Reply:
The filter uses it.  If the filter only has a udf and doesn't have any of the built in special keywords, like "height" then it doesn't use it, hence why it's not always being used.  Right now I'm not sure which keywords exactly will need it, but height definitely will.

Reply:
So when it returns none, the filter handles this, right?

Reply:
It should, yes.  Right now the keywords aren't completely implemented - that's what I'm working on after this pull request.